sigstore/_internal/fulcio/client.py

```diff
@@ -238,22 +132,12 @@ def post(
                     raise FulcioClientError(text["message"]) from http_error
             raise FulcioClientError from http_error
 
-        if resp.json().get("signedCertificateEmbeddedSct"):
-            sct_embedded = True
-            try:
-                certificates = resp.json()["signedCertificateEmbeddedSct"]["chain"][
-                    "certificates"
-                ]
-            except KeyError:
-                raise FulcioClientError("Fulcio response missing certificate chain")
-        else:
-            sct_embedded = False
```

Comment:
For instances that don’t use a CT log, the certificate chain is written to signedCertificateDetachedSct. To continue to support these local deployments, you’d need to still read the chain from either field, but only check the SCT if embedded. 

Cc @codysoyland who added that for Sigstore-go

Reply:
Is it not a problem if the server can just "downgrade" to detached sct like that? 

Or are we supposed to always require embedded SCT if the trust root contains at least one CT log, but otherwise be ok with detached SCT?

Reply:
Right now sigstore-python doesn't allow non-CT BYO uses at all, so allowing `signedCertificateDetachedSct` while removing detached SCT support would mean a larger refactor 😅 

Given that, I _think_ we're OK to remove this outright, since `signedCertificateDetachedSct`-without-CT is a use case we already didn't support. But I'm curious if this matches @jku's understanding 🙂

Reply:
> Is it not a problem if the server can just "downgrade" to detached sct like that?
> 
> Or are we supposed to always require embedded SCT if the trust root contains at least one CT log, but otherwise be ok with detached SCT?

For Sigstore-go, it’s a threshold configured by the user. For cosign, SCTs are required unless opted out explicitly by the user.

If you require CT now, then agreed this would not be a breaking change.

Reply:
@woodruffw sure, that sounds good.